contracts/token-registrars/StandardizedTokenRegistrar.sol

```diff
@@ -0,0 +1,159 @@
+        bytes32 tokenId = service.getCustomTokenId(address(this), salt);
+
+        service.deployAndRegisterStandardizedToken(salt, name, symbol, decimals, mintAmount, distributor);
+        ITokenManager tokenManager = ITokenManager(service.getTokenManagerAddress(tokenId));
+        tokenManager.transferOperatorship(sender);
+
+        IStandardizedToken token = IStandardizedToken(service.getStandardizedTokenAddress(tokenId));
+        token.safeTransfer(sender, mintAmount);
+    }
+
+    function deployRemoteStandarizedToken(
+        bytes32 salt,
+        address additionalDistributor,
+        address optionalOperator,
+        uint256 mintAmount,
+        string memory destinationChain,
+        uint256 gasValue
+    ) external payable {
+        string memory tokenName;
+        string memory tokenSymbol;
+        uint8 tokenDecimals;
+        bytes memory distributor = new bytes(0);
+        bytes memory operator = new bytes(0);
+
+        {
+            address sender = msg.sender;
+            salt = getStandardizedTokenSalt(sender, salt);
+            bytes32 tokenId = service.getCustomTokenId(address(this), salt);
+
+            IStandardizedToken token = IStandardizedToken(service.getStandardizedTokenAddress(tokenId));
+            ITokenManager tokenManager = ITokenManager(service.getTokenManagerAddress(tokenId));
+
+            tokenName = token.name();
+            tokenSymbol = token.symbol();
+            tokenDecimals = token.decimals();
+            if (additionalDistributor != address(0)) {
+                if (!token.isDistributor(additionalDistributor)) revert NotDistributor(additionalDistributor);
+                distributor = additionalDistributor.toBytes();
+            } else if (mintAmount != 0) {
+                revert NonZeroMintAmount();
+            }
+            if (optionalOperator != address(0)) {
+                if (!tokenManager.isOperator(optionalOperator)) revert NotOperator(optionalOperator);
+                operator = optionalOperator.toBytes();
+            }
+        }
+
+        _deployAndRegisterRemoteStandardizedToken(
+            salt,
+            tokenName,
+            tokenSymbol,
+            tokenDecimals,
+            distributor,
+            operator,
+            mintAmount,
+            destinationChain,
+            gasValue
+        );
+    }
+
+    function _deployAndRegisterRemoteStandardizedToken(
+        bytes32 salt,
+        string memory tokenName,
+        string memory tokenSymbol,
+        uint8 tokenDecimals,
+        bytes memory distributor,
+        bytes memory operator,
+        uint256 mintAmount,
+        string memory destinationChain,
+        uint256 gasValue
+    ) internal {
+        // slither-disable-next-line arbitrary-send-eth
+        service.deployAndRegisterRemoteStandardizedToken{ value: gasValue }(
+            salt,
+            tokenName,
+            tokenSymbol,
+            tokenDecimals,
+            distributor,
+            '',
```

Comment:
mintTo isn't being provided so I think the initial mint will be skipped? Can you add that, and add test coverage for it?

Reply:
fixed